packages/block-editor/src/components/preview-options/index.js

```diff
@@ -45,28 +48,24 @@ export default function PreviewOptions( {
 			{ () => (
 				<>
 					<MenuGroup>
-						<MenuItem
-							className="block-editor-post-preview__button-resize"
-							onClick={ () => setDeviceType( 'Desktop' ) }
-							icon={ deviceType === 'Desktop' && check }
-						>
-							{ __( 'Desktop' ) }
-						</MenuItem>
-						<MenuItem
-							className="block-editor-post-preview__button-resize"
-							onClick={ () => setDeviceType( 'Tablet' ) }
-							icon={ deviceType === 'Tablet' && check }
-						>
-							{ __( 'Tablet' ) }
-						</MenuItem>
-						<MenuItem
-							className="block-editor-post-preview__button-resize"
-							onClick={ () => setDeviceType( 'Mobile' ) }
-							icon={ deviceType === 'Mobile' && check }
-						>
-							{ __( 'Mobile' ) }
-						</MenuItem>
+						{ coreDeviceTypes.map( ( device ) => (
+							<MenuItem
+								key={ device }
+								className="block-editor-post-preview__button-resize"
+								onClick={ () => setDeviceType( device ) }
+								icon={ deviceType === device && check }
+							>
+								{ device }
+							</MenuItem>
+						) ) }
```

Comment:
I like how you introduced the `coreDeviceTypes` array and then streamlined the way the
`MenuItem` components are being rendered using `map()`.

One thing I think we could consider is to make the output of device labels
translatable.

An idea I have on my mind is to turn the array of strings into the array of objects
and let the label have its own translatable value that isn't used for any logic:

```diff
diff --git a/packages/block-editor/src/components/preview-options/index.js b/packages/block-editor/src/components/preview-options/index.js
index 03b3f3cea9..c3e67ab580 100644
--- a/packages/block-editor/src/components/preview-options/index.js
+++ b/packages/block-editor/src/components/preview-options/index.js
@@ -19,7 +19,11 @@ export default function PreviewOptions( {
 	deviceType,
 	setDeviceType,
 } ) {
-	const coreDeviceTypes = [ 'Desktop', 'Tablet', 'Mobile' ];
+	const coreDeviceTypes = [
+		{ type: 'Desktop', label: __( 'Desktop' ) },
+		{ type: 'Tablet', label: __( 'Tablet' ) },
+		{ type: 'Mobile', label: __( 'Mobile' ) },
+	];
 
 	const isMobile = useViewportMatch( 'small', '<' );
 	if ( isMobile ) return null;
@@ -50,12 +54,12 @@ export default function PreviewOptions( {
 					<MenuGroup>
 						{ coreDeviceTypes.map( ( device ) => (
 							<MenuItem
-								key={ device }
+								key={ device.type }
 								className="block-editor-post-preview__button-resize"
-								onClick={ () => setDeviceType( device ) }
-								icon={ deviceType === device && check }
+								onClick={ () => setDeviceType( device.type ) }
+								icon={ deviceType === device.type && check }
 							>
-								{ device }
+								{ device.label }
 							</MenuItem>
 						) ) }
 					</MenuGroup>
```